src/components/dataList.js

```diff
@@ -53,104 +75,150 @@
             </div>
             <div className={classes.footer}>
               {(isDev || model) && !hidePagination && (
-                <Pagination totalCount={0} resultCount={take} currentPage={1} />
+                <Pagination
+                  totalCount={0}
+                  resultCount={rowsPerPage}
+                  currentPage={1}
+                />
               )}
             </div>
           </>
         );
 
+        const handleSearch = event => {
+          setSearch(event.target.value);
+        };
+
+        const isObject = item =>
+          item && typeof item === 'object' && !Array.isArray(item);
+        const deepMerge = (target, ...sources) => {
+          if (!sources.length) return target;
+          const source = sources.shift();
+
+          if (isObject(target) && isObject(source)) {
+            // eslint-disable-next-line no-restricted-syntax
+            for (const key in source) {
+              if (isObject(source[key])) {
+                if (!target[key]) Object.assign(target, { [key]: {} });
+                deepMerge(target[key], source[key]);
+              } else {
+                Object.assign(target, { [key]: source[key] });
+              }
+            }
+          }
+
+          return deepMerge(target, ...sources);
+        };
+
+        const orderByArray = [orderBy].flat();
+        const sort = {};
+
+        if (!isDev && orderBy) {
+          orderByArray.reduce((acc, property, index) => {
+            const prop = getProperty(property);
+            // eslint-disable-next-line no-return-assign
+            return (acc[prop.name] =
+              index !== orderByArray.length - 1 ? {} : order.toUpperCase());
+          }, sort);
+        }
+
```

Comment:
This part, can be written differently using reduceRight.

```
        const sort =
          !isDev && orderBy
            ? orderByArray.reduceRight((acc, property, index) => {
                const prop = getProperty(property);
                return index === orderByArray.length - 1
                  ? { [prop.name]: order.toUpperCase() }
                  : { [prop.name]: acc };
              }, {})
            : {};

```
And you can do this in datatable as well.